packages/compile-solidity/src/index.ts

```diff
@@ -27,7 +40,7 @@ async function compileYulPaths(yulPaths, options) {
     });
     debug("Yul compiled successfully");
 
-    if (compilation.contracts.length > 0) {
+    if (compilation && compilation.contracts.length > 0) {
```

Comment:
So I'm not sure we want to throw in this instance. The `run` method returns a `Compilation` or `null` if there is nothing to compile. So if there was nothing to compile in `run`, `null` is returned and we don't push the result into the array.

Reply:
So I'll just respond to all the comments of this type here. I think the old behavior is buggy. Should it return `null` it would throw in the old code as you can't access `contracts` on `null`. So I was considering this fixing a bug.

Reply:
Yeah if you're confident that everything works fine this way, then it's fine by me.